file_catalog/server.py

```diff
@@ -1,51 +1,61 @@
+# server.py
 """File Catalog REST Server Interface."""
 
 # fmt: off
 # pylint: disable=R0913,R0903
 
-from __future__ import absolute_import, division, print_function
-
-import copy
 import datetime
 import logging
 import os
 import random
 import sys
-from collections import OrderedDict
-from importlib.abc import Loader
 from pkgutil import get_loader
-from typing import Any, Dict, Optional, cast
+from typing import Any, Callable, Dict, Optional, cast
 from uuid import uuid1
 
-import pymongo.errors  # type: ignore[import]
-import tornado.ioloop
-import tornado.web
-from rest_tools.server import RestHandler, handler
-from rest_tools.utils import Auth
+from rest_tools.server import keycloak_role_auth, RestHandler, RestHandlerSetup, RestServer
 from tornado.escape import json_decode, json_encode
-from tornado.httputil import url_concat
-
-# local imports
-import file_catalog
 
-from . import argbuilder, deconfliction, urlargparse
+from . import argbuilder, deconfliction, urlargparse2
 from .mongo import Mongo
 from .schema import types
 from .schema.validation import Validation
 
-logger = logging.getLogger('server')
+logger = logging.getLogger(__name__)
 
 StrDict = Dict[str, Any]
 
+CONFIG_LOGGING_DENY_LIST = ["FC_COOKIE_SECRET", "MONGODB_AUTH_PASS"]
+
+FC_AUTH_PREFIX = "resource_access.file-catalog.roles"
+FC_AUTH_ROLES = ["system"]
+
 
 # --------------------------------------------------------------------------------------
-# Utils
+# Auth
 # --------------------------------------------------------------------------------------
 
+if bool(os.environ.get('CI_TEST_ENV', False)):
```

Comment:
This will be true for all values of `CI_TEST_ENV`. You may want to use `wipac-dev-tools`' `strtobool()`

Reply:
I'm personally fine with this, since you shouldn't define the env variable at all in prod.  But yeah, the conversion from str to bool is technically wrong (and unnecessary, since it automatically happens in the conditional).

Reply:
Good catch; I changed this to a simple "if present" check, no bool'ing.

Reply:
I'll take it :smile: